Storage/Storage/CoreData/CoreDataManager.swift

```diff
@@ -37,21 +39,38 @@ public class CoreDataManager: StorageManagerType {
         container.persistentStoreDescriptions = [storeDescription]
 
         container.loadPersistentStores { [weak self] (storeDescription, error) in
-            guard let `self` = self, let error = error else {
+            guard let `self` = self, let errorLoadingPersistentStores = error else {
                 return
             }
 
-            DDLogError("⛔️ [CoreDataManager] loadPersistentStore failed. Attempting to recover... \(error)")
+            DDLogError("⛔️ [CoreDataManager] loadPersistentStore failed. Attempting to recover... \(errorLoadingPersistentStores)")
 
             /// Backup the old Store
             ///
             do {
                 let sourceURL = self.storeURL
                 let backupURL = sourceURL.appendingPathExtension("~")
                 try FileManager.default.copyItem(at: sourceURL, to: backupURL)
-                try FileManager.default.removeItem(at: sourceURL)
             } catch {
-                fatalError("☠️ [CoreDataManager] Cannot backup Store: \(error)")
+                let message = "☠️ [CoreDataManager] Cannot backup Store: \(error)"
+                self.crashLogger.logMessage(message,
+                                            properties: ["loadPersistentStoresError": errorLoadingPersistentStores,
+                                                         "backupError": error],
```

Comment:
great catch! I looked into [where this error is called](https://github.com/getsentry/sentry-cocoa/blob/fb4b4be84ae61f9e306348a65600c9dd93977e6f/Sources/Sentry/SentryNSURLRequest.m#L53-L58), and it's because the dictionary for the additional data can't be serialized to JSON - `NSError` isn't serializable via `JSONSerialization` unfortunately.

The [`CrashLogging.logError`](https://github.com/Automattic/Automattic-Tracks-iOS/blob/18852dacf9385d4fe8d5b612c206f135ba6b8606/Automattic-Tracks-iOS/Crash%20Logging/CrashLogging.swift#L110) call has the same issue as `CrashLogging.logMessage` because it's also setting the same dictionary to `Event`'s `extra: [String: Any]` field.

I followed [how WPiOS serialized the `NSError`](https://github.com/wordpress-mobile/WordPress-iOS/blob/9f25996449e2cddb387f947b4d765053e07cabf0/WordPress/Classes/Utility/Logging/SentryStartupEvent.swift#L40-L44) in this commit https://github.com/woocommerce/woocommerce-ios/pull/2366/commits/781b0f6b015f68439b316a4d81bed279d3b84007 so that it keeps all the serializable values and serializing values of `NSError` type for now. Let me know what you think!

Reply:
good idea! yea it'd be helpful to understand what caused the error in the first place. added logging in: https://github.com/woocommerce/woocommerce-ios/pull/2366/commits/4a6709410c65b90d6e0768796efeee0775738190